test/models/pay/subscription_test.rb

```diff
@@ -118,6 +118,28 @@ class Pay::Subscription::Test < ActiveSupport::TestCase
     refute_includes subscriptions, subscription5
   end
 
+  test "with_active_customer scope" do
+    subscription = create_subscription
+    subscriptions = Pay::Subscription.with_active_customer
+
+    assert_includes subscriptions, subscription
+
+    @pay_customer.update!(deleted_at: Time.now)
+
+    refute_includes subscriptions, subscription
+  end
+
+  test "with_deleted_customer scope" do
+    subscription = create_subscription
+    subscriptions = Pay::Subscription.with_deleted_customer
+
+    refute_includes subscriptions, subscription
+
+    @pay_customer.update!(deleted_at: Time.now)
+
+    assert_includes subscriptions, subscription
```

Comment:
You're saving the results of the query, so the second assert here doesn't re-run the query.

Instead you want two fresh queries:
```ruby
    refute_includes Pay::Subscription.with_deleted_customer, subscription

    @pay_customer.update!(deleted_at: Time.now)

    assert_includes Pay::Subscription.with_deleted_customer, subscription
```